internal/engine/upper.go

```diff
@@ -786,7 +800,16 @@ func handleLocalServeReadinessProbeAction(ctx context.Context, state *store.Engi
 	}
 
 	lrs := ms.LocalRuntimeState()
-	lrs.SetReadinessProbeState(action.Ready)
+	lrs.Ready = action.Ready
+	lrs.LastReadyOrSucceededTime = time.Now()
```

Comment:
this should only happen if `Ready` is `true`, right?

Reply:
Fixed in 4eee419